source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/EmissionValueKey.java

```diff
@@ -185,8 +194,35 @@ public void setSubstance(final Substance substance) {
     this.substance = substance;
   }
 
+  /**
+   * Converts this EmissionValueKey to a parseable string representation in the
+   * format "year:substance".
+   * This format is compatible with {@link #fromStringValue(String)} for
+   * serialization/deserialization.
+   * 
+   * @return String representation in format "year:substance"
+   * @see #fromStringValue(String)
+   */
+  public String toStringValue() {
+    return year + ":" + substance.name();
+  }
+
+  /**
+   * Creates an EmissionValueKey from its string representation.
+   * Expects input in the format "year:substance" as produced by
+   * {@link #toStringValue()}.
+   * 
+   * @param value The string representation in format "year:substance"
+   * @return A new EmissionValueKey instance
+   * @see #toStringValue()
+   */
+  public static EmissionValueKey fromStringValue(final String value) {
+    final String[] parts = value.split(":");
+    return new EmissionValueKey(Integer.parseInt(parts[0]), Substance.valueOf(parts[1]));
```

Comment:
Won't this fail if you have an unexpected string as input like `"whoops"`?

Reply:
("me")